### PR TITLE
Validations, specs and n+1

### PR DESCRIPTION
**Pull request by AlexRS90**
For this PR I:
✅  Added the following validations:
> ✔️  For the User model:
>               ☑️  Name must not be blank.
>               ☑️  PostsCounter must be an integer greater than or equal to zero.
> ✔️  For the Post model:
>               ☑️  Title must not be blank.
>               ☑️  Title must not exceed 250 characters.
>               ☑️  CommentsCounter must be an integer greater than or equal to zero.
>               ☑️  LikesCounter must be an integer greater than or equal to zero.

✅  Added unit specs for all of your models' methods and validations.
✅  Added flash messages in the create actions in all your controllers.
✅   Solved issue N+1 using `Bullet gem`

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -17,13 +17,14 @@ def new
...
       flash[:success] = 'New post created successfully!'
-      redirect_to users_path
+      redirect_to "/users/#{@users.id}/posts"
     else
-      flash.now[:error] = 'Error: Data was not saved!'
-      render :new
+      flash[:error] = @post.errors.full_messages[0]
+      redirect_to "/users/#{@users.id}/posts/new"
     end
   end
```
Comment (DeliceLydia) [inline app/controllers/posts_controller.rb:30]: - [ ] I suggest that you solve the n+1 problem when fetching all posts and their comments  you can simply add in your index method `@posts = @user.posts.includes(:comments)` currently to load a user posts page and related comments is a problem
query

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @AlexRS90, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Major highlights for your project:
- Good README
- No linter errors
- All your routes and validations are working well

## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (AlexRS90) [reply app/controllers/posts_controller.rb:30]: @DeliceLydia  Thank you very much for the feedback, I'll work on it!

**Review by DeliceLydia — APPROVED** [review]
# Congratulations! 🎉🎉 

## Hi @AlexRS90, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AlexRS90** [comment]
@DeliceLydia  thank you very much!!

**Pull request merged**